File: baekjoon/src/tags/queue/No18258.java
package tags.queue;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;

/**
 * https://www.acmicpc.net/problem/18258
 */
public class No18258 {
    public static final Deque<Integer> deque = new LinkedList<>();

    public static final StringBuilder builder = new StringBuilder();

    public static void command(String command) {
        String[] orders = command.split(" ");
        if (orders.length == 1) {
            switch (orders[0]) {
                case "pop":
                    builder.append(deque.isEmpty() ? -1 : deque.pollFirst()).append("\n");
                    break;
                case "size":
                    builder.append(deque.size()).append("\n");
                    break;
                case "empty":
                    builder.append(deque.isEmpty() ? 1 : 0).append("\n");
                    break;
                case "front":
                    builder.append(deque.isEmpty() ? -1 : deque.peekFirst()).append("\n");
                    break;
                case "back":
                    builder.append(deque.isEmpty() ? -1 : deque.peekLast()).append("\n");
                    break;
            }
        } else
            deque.add(Integer.parseInt(orders[1]));
    }

    public static void printResult() {
        System.out.println(builder);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int numberOfInput = Integer.parseInt(reader.readLine());
        for (int i = 0; i < numberOfInput; i++) {
            String command = reader.readLine();
            command(command);
        }
        printResult();
    }
}

File: baekjoon/src/tags/data_structures/No18258.kt
package tags.data_structures

/**
 * https://www.acmicpc.net/problem/18258
 */
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val reader = BufferedReader(InputStreamReader(System.`in`))
private val writer = BufferedWriter(OutputStreamWriter(System.out))

fun main() {
    val queue = ArrayDeque<Int>()
    val n = reader.readLine().toInt()
    repeat(n) {
        val s = reader.readLine().split(" ")
        when (s[0]) {
            "push" -> queue.add(s[1].toInt())
            "pop" -> writer.write("${queue.removeFirstOrNull() ?: -1}\n")
            "size" -> writer.write("${queue.size}\n")
            "empty" -> if (queue.isEmpty()) {
                writer.write("1\n")
            } else {
                writer.write("0\n")
            }

            "front" -> writer.write("${queue.firstOrNull() ?: -1}\n")
            "back" -> writer.write("${queue.lastOrNull() ?: -1}\n")
        }
    }
    writer.flush()
}
